transcom/mymove
# MB-3619 Add queries to MoveTaskOrder page

## Description

> On the MoveTaskOrder page, replace the existing data fetch code with React Query, to fetch and populate MTO shipments and MTO service items.
> 
> Notes:
> 
> MoveTaskOrder should be converted from a class component to a functional component
> 
> The queries triggered by componentWillMount should be replaced with useQuery hooks, and Redux connect code should be deleted
> 
> The logic in existing mapStateToProps that maps API data to the format expected by the component may need to be replicated in the useQuery hook.
> 
> This may be a good time to also move the MoveTaskOrder.jsx file into the new file structure (ie., pages/Office/MoveTaskOrder/MoveTaskOrder.jsx)
> 
> AC:
> 
> MoveTaskOrder does not rely on Redux for any API data
> 
> MoveTaskOrder queries the API directly for any data needed (MTO shipments & MTO service items) based on the MTO ID in the URL params

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

To test:

1. Login in to Office app as TOO, local
2. Navigate to http://officelocal:3000/moves/5d4b25bb-eb04-4c03-9a81-ee0398cb779e/mto
3. mtoShipments and mtoServiceItems should be loaded

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3619) for this change
* [Article about react-query](https://react-query.tanstack.com/docs/api#usequery)
